src/app/users/store/users.reducer.ts

```diff
@@ -1,41 +1,51 @@
 import { IUser } from '../user.interface';
 import { createReducer, on, Action } from '@ngrx/store';
 import { getAllUsersActions } from './users.actions';
+import { createEntityAdapter, EntityAdapter, EntityState } from '@ngrx/entity';
+import { state } from '@angular/animations';
 
+//export interface IUsersInitialState extends EntityState<IUser> {
 export interface IUsersInitialState {
   users: IUser[];
-  getAllUsersPending: boolean;
-  getAllUsersError: any;
 }
+//export const userAdapter: EntityAdapter<IUser> = createEntityAdapter<IUser>();
 export const usersInitialState: IUsersInitialState = {
   users: [],
-  getAllUsersPending: false,
-  getAllUsersError: null,
 };
+//userAdapter.getInitialState({
+
+//users: [],
+//  selectedUserId: null,
+//});
 
 const userReducer = createReducer(
   usersInitialState,
   on(getAllUsersActions.begin, (state) => ({
     ...state,
-    getAllUsersPending: true,
```

Comment:
why did you remove all this? This is de essence of async functionality. With this, you can show errors of activity indication to the end-users of your app!

Reply:
I replaced it with an error message,. please let me know if you agree

Reply:
OK you you dont want to use de store for the error (but is a good practice to program that).  but de pending flag is a must that have to be there.